Sources/RRule.swift

```diff
@@ -181,22 +179,33 @@ public class RRule: NSObject {
       return nil
     }
     recurrenceRule.frequency = frequency
+    guard recurrenceRule.frequency == .monthly &&
+            recurrenceRule.bymonthday.isEmpty &&
+            recurrenceRule.bysetpos.isEmpty &&
+            !byDayValue.isEmpty
+    else { return recurrenceRule }
     
+    let offsetString = byDayValue.components(separatedBy: CharacterSet.decimalDigits.inverted).joined()
+    if offsetString.isEmpty {
+      return recurrenceRule
+    }
+    var offset = Int(offsetString)!
+    if byDayValue.starts(with: "-") {
+      offset = -1 * offset
+    }
+    recurrenceRule.bysetpos = [offset]
+    recurrenceRule.byweekday = [EKWeekday.weekdayFromSymbol(String(byDayValue.suffix(2)))!]
     return recurrenceRule
   }
   
   public static func stringFromRule(_ rule: RecurrenceRule) -> String {
```

Comment:
do we not need to revert string parsing we did for the monthly recurring events bug in this function for cross platform use